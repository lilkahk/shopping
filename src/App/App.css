.welcome-front-page {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 4rem 3rem 2rem;
    gap: 6px;
}

.welcome-front-page h1 {
    font-size: 4rem;
}

.welcome-front-page h3 {
    font-size: 2rem;
    font-weight: 500;
}

.welcome-btns {
    display: flex;
    gap: 1rem;
    padding-top: 1rem;
}

.welcome-btns button {
    outline: none;
    border-radius: 16px;
    padding: 0.5rem 1rem;
    width: 188px;
    font-size: 1rem;
    text-align: center;
    cursor: pointer;
    transition: scale 0.1s ease-in-out, background-color 0.1s ease-in-out;
}

.welcome-btns button:hover {
    scale: 1.05;
}

.shop-now-btn {
    color: white;
    background-color: var(--main-bck-col);
    border: none;
}

.shop-now-btn:hover {
    background-color: var(--main-bck-dark);
}

.see-collections-btn {
    border: 2px solid var(--main-bck-col);
    color: var(--main-bck-col);
    background-color: white;
}

.app-card {
    box-sizing: border-box;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 2rem;
}
