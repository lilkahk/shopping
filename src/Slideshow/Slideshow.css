.image-slideshow {
    object-fit: cover;
    width: 100%;
    height: 100%;
    display: block;
    flex-shrink: 0;
    flex-grow: 0;
    transition: translate 0.27s ease-in-out;
}

.image-slideshow-btn {
    all: unset;
    display: block;
    position: absolute;
    top: 0;
    bottom: 0;
    cursor: pointer;
    padding: 1rem;
    transition: background-color 0.12s ease-in-out;
}

.image-slideshow-btn:hover, .image-slideshow-btn:focus-visible {
    background-color: rgba(0, 0, 0, 0.291);
}

.image-slideshow-btn > * {
    stroke: white;
    color: rgba(255, 255, 255, 0.652);
}

.image-slideshow-btn:hover > *, .image-slideshow-btn:focus-visible > * {
    animation: button-animate 0.21s ease-in-out;
}

@keyframes button-animate {
    50% {
        scale: 1.4 0.6;
    }
}

.dot-btn {
    all: unset;
    display: block;
    cursor: pointer;
    width: 0.8rem;
    height: 0.8rem;
}

.div-dot-btn, .div-dot-btn-selected {
    height: 100%;
    width: 100%;
    border: 2px solid rgb(1, 1, 1);
    background-color: rgba(0, 0, 0, 0.379);
    border-radius: 1rem;
    transition: scale 0.1s ease-in-out;
}

.div-dot-btn:hover, .div-dot-btn:focus-visible,
.div-dot-btn-selected:hover, .div-dot-btn-selected:focus-visible {
    scale: 1.2;
}

.div-dot-btn-selected {
    border: 2px solid white;
    background-color: rgba(255, 255, 255, 0.608);
}

.dot-btn:focus-visible, .image-slideshow-btn:focus-visible {
    outline: auto;
}

.skip-link {
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    border: 0;
    clip: rect(0, 0, 0, 0);
}

.skip-link:focus-visible {
    top: 0;
    left: 0;
    border: 1px solid black;
    background-color: white;
    padding: 0.5rem;
    width: auto;
    height: auto;
    margin: 0;
    clip: unset;
    text-decoration: none;
    color: black;
    z-index: 100;
}