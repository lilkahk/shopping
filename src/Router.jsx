import { RouterProvider, createBrowserRouter } from "react-router-dom";
import App from "./App/App";
import Layout from "./Layout/Layout";
import Collections from "./Collections/Collections";
import AboutUs from "./AboutUs/AboutUs";
import ContactUs from "./ContactUs/ContactUs";
import OurStory from "./OurStory/OurStory";
import AllProducts from "./AllProducts/AllProducts";
import Mens from "./Mens/Mens";
import Womens from "./Womens/Womens";
import Jewelry from "./Jewelry/Jewelry";

const Router = () => {
    const router = createBrowserRouter([
        {
          path: "/",
          element: <Layout />,
          children: [
            {
              index: true,
              element: <App />
            },
            {
              path: "collections",
              element: <Collections />
            },
            {
              path: "mens",
              element: <Mens />
            },
            {
              path: "womens",
              element: <Womens />
            },
            {
              path: "jewelry",
              element: <Jewelry />
            },
            {
              path: "about-us",
              element: <AboutUs />
            },
            {
              path: "contact-us",
              element: <ContactUs />
            },
            {
              path: "our-story",
              element: <OurStory />
            },
            {
              path: "all-products",
              element: <AllProducts />
            }
          ]
        },
      ]);
    return <RouterProvider router={router} />
}

export default Router;