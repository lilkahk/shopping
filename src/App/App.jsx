import './App.css'
import Slideshow from '../Slideshow/Slideshow'
import { Link } from 'react-router-dom'
import DescriptionCard from '../DescriptionCard/DescriptionCard'
import firstCard from './images/first-card.jpg'
import ThreeColumn from '../ThreeColumn/ThreeColumn'
import quality from './images/quality.png'
import delivery from './images/delivery.png'
import customer from './images/customer-service.png'
import pic1 from './slides/one.png'
import pic2 from './slides/two.jpg'
import pic3 from './slides/three.png'
import pic4 from './slides/four.jpg'
import pic5 from './slides/five.jpg'

export const SLIDES = [
  {url: pic1, alt: 'Slide 1: 3 friends having fun'},
  {url: pic2, alt: 'Slide 2: 2 friends smiling at the camera'},
  {url: pic3, alt: 'Slide 3: 4 friends having fun at a party'},
  {url: pic4, alt: 'Slide 4: 5 friends smiling while camping at night'},
  {url: pic5, alt: 'Slide 5: 4 girls smiling while sitting down together'}
]

const App = () => {

  return (
    <>
      <div style={{width: '100%', height: '600px'}}>
            <Slideshow slides={SLIDES} />
      </div>
      <WelcomeTitle />
      <div className='app-card'>
        <DescriptionCard
          url={firstCard}
          alt={'Friends having fun while the sun sets in the distance'} 
          title={'Clothing & More'} 
          text={
            [
              "Dive into a world of endless possibilities as you explore our \
              carefully curated collection of clothing and jewelry",
              "Discover the perfect pieces that will make you stand out and feel confident."
            ]
          }
          link={'all-products'}
          buttonText={'SHOP NOW'}
        />
      </div>
      <div className='app-card'>
        <ThreeColumn title={'Why Us?'} data={
          [
            {
              url: quality,
              alt: 'Great Quality Guaranteed Icon',
              title: 'Great Quality Guaranteed',
              text: 'Our clothing and jewelry embody unparalleled quality,\
                exuding sophistication and excellence. Choose us for a touch of\
                luxury that defines your uniqueness',
            },
            {
              url: delivery,
              alt: 'Fast Shipping Icon',
              title: 'Fast Shipping',
              text: 'Our fast shipping promise: Less than 3 days within America\
                and under a week worldwide. Choose us for swift delivery, ensuring\
                your satisfaction knows no bounds',
            },
            {
              url: customer,
              alt: '24h Customer Service Icon',
              title: '24h Customer Service',
              text: 'Our 24-hour customer service commitment ensures your needs\
                are met day and night. Choose us for swift assistance, putting\
                your satisfaction at the heart of our service',
            },
          ]
        } />
      </div>
    </>
  )
};

export default App

const WelcomeTitle = () => {
  return (
    <section className='welcome-front-page'>
      <h1>Welcome to My<em>SHOP</em></h1>
      <h3>The best store out there</h3>
      <div className='welcome-btns'>
        <Link to='all-products'>
          <button className='shop-now-btn'>SHOP NOW</button>
        </Link>
        <Link to='collections'>
          <button className='see-collections-btn'>SEE COLLECTIONS</button>
        </Link>
      </div>
    </section>
  )
}
