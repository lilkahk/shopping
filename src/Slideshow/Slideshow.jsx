import './Slideshow.css'
import { useEffect, useState } from 'react'
import Icon from '@mdi/react';
import { mdiArrowLeftBold } from '@mdi/js';
import { mdiArrowRightBold } from '@mdi/js';
import PropTypes, { objectOf, string } from 'prop-types';

const Slideshow = ({ timer = 5000, slides }) => {

    const SLIDES = slides

    const [imageIdx, setImageIdx] = useState(0)

    const showPrevious = () => {
        setImageIdx(prev => (prev - 1 + SLIDES.length) % SLIDES.length)
    }

    const showNext = () => {
        setImageIdx(prev => (prev + 1) % SLIDES.length)
    }

    useEffect(() => {
        const interval = setInterval(() => {
        setImageIdx(prev => (prev + 1) % SLIDES.length);
        }, timer);

        return () => {
        clearInterval(interval)
        }
    }, [SLIDES, timer])

    return (
        <section aria-label='Image Slideshow' style={{ width: '100%', height: '100%', position: 'relative' }}>
            <a href='#skip-slideshow' className='skip-link'>Skip slideshow</a>
            <div style={{ width: '100%', height: '100%', overflow: 'hidden', display: 'flex'}}>
                {SLIDES.map(( { url, alt }, index) => {
                    return <img
                             key={url} src={url} alt={alt}
                             className='image-slideshow' aria-hidden={index !== imageIdx}
                             style={{ translate: `${-100 * imageIdx}%`}} />
                })}
            </div>
            {/* <img src={SLIDES[imageIdx]} className='image-slideshow' /> */}
            <button className='image-slideshow-btn' onClick={showPrevious}
            style={{ left: '0', paddingRight: '1.5rem' }} aria-label='View Previous Image'>
                <Icon path={mdiArrowLeftBold} size={1.5} />
            </button>
            <button className='image-slideshow-btn' onClick={showNext}
                style={{ right: '0', paddingLeft: '1.5rem' }} aria-label='View Next Image'>
                <Icon path={mdiArrowRightBold} size={1.5} />
            </button>
            <div style={{ position: 'absolute', bottom: '0.5rem', left: '50%', 
            translate: '-50%', display: 'flex', gap: '0.7rem', paddingBottom: '6px' }}>
                {SLIDES.map((slide, index) => {
                    return <button key={index} className='dot-btn'
                         onClick={() => setImageIdx(index)} aria-label={`View Image ${index + 1}`}>
                        <div className={index === imageIdx ? 'div-dot-btn-selected' : 'div-dot-btn'}></div>
                    </button>
                })}
            </div>
            <div id='skip-slideshow'> </div>
        </section>
    )
}

Slideshow.propTypes = {
    timer: PropTypes.number,
    slides: PropTypes.arrayOf(objectOf(string))
}

export default Slideshow