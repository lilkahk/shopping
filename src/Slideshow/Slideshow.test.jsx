import { render, screen } from "@testing-library/react"
import { beforeEach, describe, expect, it } from "vitest"
import { userEvent } from "@testing-library/user-event"
import Router from "../Router"
import Slideshow from "./Slideshow"
import { act } from "react-dom/test-utils"
import { SLIDES } from "../App/App"

// export const SLIDES = [
//     {url: pic1, alt: 'Slide 1: 3 friends having fun'},
//     {url: pic2, alt: 'Slide 2: 2 friends smiling at the camera'},
//     {url: pic3, alt: 'Slide 3: 4 friends having fun at a party'},
//     {url: pic4, alt: 'Slide 4: 5 friends smiling while camping at night'},
//     {url: pic5, alt: 'Slide 5: 4 girls smiling while sitting down together'}
//   ]

describe("slideshow", () => {
    beforeEach(() => {
        render(<Router />)
    })

    function checkNotHidden(idx) {

        const allSlides = 
        [
            screen.getByAltText(SLIDES[0].alt),
            screen.getByAltText(SLIDES[1].alt),
            screen.getByAltText(SLIDES[2].alt),
            screen.getByAltText(SLIDES[3].alt),
            screen.getByAltText(SLIDES[4].alt)
        ]
        
        allSlides.forEach((image, index) => {
            if (index === idx) {
            expect(image).toHaveAttribute('aria-hidden', 'false')
            } else {
            expect(image).toHaveAttribute('aria-hidden', 'true')
            }
        })
    }

    it("renders first image", () => {
        const firstImage = screen.getByAltText(SLIDES[0].alt)
        const secondImage = screen.getByAltText(SLIDES[1].alt)
        expect(firstImage).toHaveAttribute('aria-hidden', 'false')  
        expect(secondImage).toHaveAttribute('aria-hidden', 'true')             
    })

    it("next and previous buttons work", async () => {
        const nextButton = screen.getByLabelText('View Next Image')
        const prevButton = screen.getByLabelText('View Previous Image')
        const user = userEvent.setup()

        checkNotHidden(0)

        await user.click(nextButton)
        checkNotHidden(1)

        await user.click(nextButton)
        checkNotHidden(2)

        await user.click(prevButton)
        checkNotHidden(1)

        await user.click(prevButton)
        checkNotHidden(0)
    })

    it('check loop around function works', async () => {
        const nextButton = screen.getByLabelText('View Next Image')
        const prevButton = screen.getByLabelText('View Previous Image')
        const user = userEvent.setup()

        await user.click(prevButton)
        checkNotHidden(4)

        await user.click(prevButton)
        checkNotHidden(3)

        await user.click(nextButton)
        checkNotHidden(4)

        await user.click(nextButton)
        checkNotHidden(0)
    })

    it('check if dots work', async () => {
        const button0 = screen.getByLabelText('View Image 1')
        const button1 = screen.getByLabelText('View Image 2')
        const button2 = screen.getByLabelText('View Image 3')
        const button3 = screen.getByLabelText('View Image 4')
        const button4 = screen.getByLabelText('View Image 5')
        const user = userEvent.setup()

        await user.click(button2)
        checkNotHidden(2)

        await user.click(button4)
        checkNotHidden(4)

        await user.click(button1)
        checkNotHidden(1)

        await user.click(button3)
        checkNotHidden(3)

        await user.click(button0)
        checkNotHidden(0)
    })
})

function sleep(ms) {
    return new Promise((resolve) => setTimeout(resolve, ms))
}

describe("automatic scroll", () => {
    it("custom timer", async () => {
      const timer = 20;

      function checkNotHidden(idx) {
        const allSlides = [
          screen.getByAltText(SLIDES[0].alt),
          screen.getByAltText(SLIDES[1].alt),
          screen.getByAltText(SLIDES[2].alt),
          screen.getByAltText(SLIDES[3].alt),
          screen.getByAltText(SLIDES[4].alt),
        ];
        allSlides.forEach((image, index) => {
          if (index === idx) {
            expect(image).toHaveAttribute("aria-hidden", "false")
          } else {
            expect(image).toHaveAttribute("aria-hidden", "true")
          }
        });
      }
  
      await act(async () => {
        render(<Slideshow timer={timer} slides={SLIDES} />)
      })

        await act(() => sleep(timer))
        checkNotHidden(1);
        await act(() => sleep(timer))
        checkNotHidden(2);
        await act(() => sleep(timer))
        checkNotHidden(3);
        await act(() => sleep(timer))
        checkNotHidden(4);
        await act(() => sleep(timer))
        checkNotHidden(0);
    });
})
  
  
  
  
  