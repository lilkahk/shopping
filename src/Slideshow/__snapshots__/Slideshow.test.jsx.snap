// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`automatic scroll > custom timer 1`] = `<div />`;

exports[`automatic scroll > custom timer 2`] = `<div />`;

exports[`automatic scroll > custom timer 3`] = `<div />`;
